_posts/2018-02-15-Rust-1.24.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: "Announcing Rust 1.24"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce a new version of Rust, 1.24.0. Rust is a
+systems programming language focused on safety, speed, and concurrency.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.24.0 is as easy as:
+
+```bash
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.24.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1240-2018-02-15
+
+## What's in 1.24.0 stable
+
+This release contains two very exciting new features: `rustfmt` and incremental compilation!
+
+#### `rustfmt`
+
+For years now, we've wanted a tool that automatically can reformat your Rust code to some sort
+of "standard style." With this release, we're happy to announce that a *preview* of `rustfmt`
+can be used with 1.24 stable. Before we get into the details, here's how to install `rustfmt`:
+
+```bash
+$ rustup component add rustfmt-preview
+```
+
+Notably, you should *not* install `rustfmt` through `cargo install`. Why not? Well, `rustfmt`
+relies on using the compiler as a library to do its job. As such, the exact right version of
+`rustfmt` for the exact compiler version you're using is critical, otherwise, it will not work.
+`rustup` knows what compiler version you're using, and so is able to pair it with the correct
+`rustfmt` version. Once you've added this component to a given toolchain, `rustup` will
+automatically update it as well. Run the command once, and then on every future
+`rustup update stable`, `rustfmt-preview` will be updated as well.
+
+> This strategy is expected to be used for other developer tools that integrate tightly with
+> the compiler, like the RLS and Clippy, in the future.
```

Comment:
We already do this for RLS

Reply:
yeah, but we've never talked about it before, so i phrased it that way